halo2_frontend/src/dev/cost_model.rs

```diff
@@ -362,7 +385,14 @@ pub fn from_circuit_to_cost_model_options<F: Ord + Field + FromUniformBytes<64>,
         (rows_count, table_rows_count, compressed_rows_count)
     };
 
-    let k = prover.k.try_into().unwrap();
+    let min_k = [
+        rows_count + cs.blinding_factors(),
+        table_rows_count + cs.blinding_factors(),
+        instance_len,
```

Comment:
It could be nice to output a warning if instance_len is the leading term here. That would mean the verifier is doing a linear amount of work, which is probably a design mistake.